be/src/formats/parquet/group_reader.cpp

```diff
@@ -274,6 +275,14 @@ Status GroupReader::_create_column_reader(const GroupReaderParam::Column& column
             RETURN_IF_ERROR(ColumnReader::create(_column_reader_opts, schema_node, column.slot_type(),
                                                  column.t_iceberg_schema_field, &column_reader));
         }
+        if (column_reader == nullptr) {
+            if (column.t_iceberg_schema_field == nullptr) {
+                return Status::InternalError("Invalid file: No valid column reader.");
```

Comment:
@Samrose-Ahmed maybe it will be better that split the bugfix and enhancement, as i know, we only materialize column that has at least one subfiled in parquet reader, for a padding column, it's padded at scanner level. cc @Smith-Cruise

Reply:
Yes its for struct evolution in Iceberg e.g. you had `event: struct<action: string>` and you evolved schema to `event: struct<action: string, id:string>`, according to Iceberg spec, if you select `event.id` you need to pad the old files that don't have `event.id` column with null.

Reply:
Yes, for this case, we want to query event.id and the former files don't contains this subfield, i actually think we have covered this case, and treat the column as column that not need materialized in the reader.

Reply:
I debugged and it crashes on one of our Iceberg tables. I added a unit test to replicate.